### DISCUSSION TITLE
useSignMessage signs different messages on localhost vs prod.

### DISCUSSION THREAD

**Discussion by jiyuhan** [wevm/wagmi#2987] [Q&A]
hey, i'm writing something that requests user to sign a message, something like

```ts

// in a component
...
const { data, signMessage } = useSignMessage();

const onButtonClick = () => {
  signMessage({ message: someMessage });
}

return <button onClick={() => {
  logic();
  onButtonClick();
  logic();
}};

...
```

so on localhost, this works fine. metamask popped up with a message showing:
```
Message:
Welcome to the new beginning
```

which is perfect.

But I have a deployed website on vercel (Nextjs) and Metamask pop-up says

```
Message:
0x0e87346afa9e0f65ba5791b77ff74fa36f347e51902c44c401480cfb763be7c3
```

Any idea why this is happening?

**Comment by adrianotadao** [comment]
hello there,

> so on localhost, this works fine. metamask popped up with a message showing:

not sure what is missing from my side but this does not happen while signing the message, any idea?

**Comment by jiyuhan** [comment]
ok just to follow up, wagmi was working as expected. this was happening because we used an account abstraction wallet which is owned by a MetaMask wallet, thus causing the confusion mentioned in the post.
